### PR TITLE
Rpm package

### LINKED ISSUE #109: Fedora packages needed
I'm thinking of setting up a copr repository for hs-notmuch and purebred in order to start using it as a more real life user experience.

### PR DESCRIPTION
**Pull request by romanofski**
Closes #109

### CODE REVIEW

**Comment by frasertweedale** [comment]
Shouldn't there be a builddep on `c2hs`?

**Comment by romanofski** [comment]
Ah actually... that's in the spec file for hs-notmuch. See didn't create a pull request for that one.

**Comment by frasertweedale** [comment]
Aah, yes of course.  That's fine then.

File: src/UI/Actions.hs [hunk 1]
```diff
@@ -507,10 +508,13 @@ invokeEditor' s = do
...
-  status <- system (editor <> " " <> tmpfile)
+  status <- onException (system (editor <> " " <> tmpfile)) (pure $ setError editorError)
   case status of
-    ExitFailure _ -> pure $ set asAppMode BrowseMail s -- ^ show error XXX
-    ExitSuccess -> pure $ set asAppMode ComposeEditor s & asCompose . cTmpFile ?~ tmpfile -- ^ go to compose editor
+    ExitFailure _ -> pure $ s & set asAppMode BrowseMail & setError editorError
+    ExitSuccess -> pure $ set asAppMode ComposeEditor s & asCompose . cTmpFile ?~ tmpfile -- go to compose editor
+
+editorError :: Error
+editorError = GenericError "Editor command exited with error code. Check the your editor configuration and your terminal."
```
Comment (frasertweedale) [inline src/UI/Actions.hs]: "Check the your editor" -> "Check your editor".

Also could you wrap this long line?

File: src/UI/Actions.hs [hunk 2]
```diff
@@ -507,10 +508,13 @@ invokeEditor' s = do
   let editor = view (asConfig . confEditor) s
   tmpdir <- getTemporaryDirectory
   tmpfile <- emptyTempFile tmpdir "purebred.tmp"
-  status <- system (editor <> " " <> tmpfile)
+  status <- onException (system (editor <> " " <> tmpfile)) (pure $ setError editorError)
   case status of
-    ExitFailure _ -> pure $ set asAppMode BrowseMail s -- ^ show error XXX
-    ExitSuccess -> pure $ set asAppMode ComposeEditor s & asCompose . cTmpFile ?~ tmpfile -- ^ go to compose editor
+    ExitFailure _ -> pure $ s & set asAppMode BrowseMail & setError editorError
+    ExitSuccess -> pure $ set asAppMode ComposeEditor s & asCompose . cTmpFile ?~ tmpfile -- go to compose editor
```
Comment (frasertweedale) [inline src/UI/Actions.hs]: If we're editing this line could we at the same time move from
```haskell
asCompose . cTmpFile ?~ tmpfile 
```
to
```haskell
set (asCompose . cTmpFile) (Just tmpfile)
```
The symbolic lens functions are hard to read.
I can never remember what they mean; what hope do newbies have?

Comment (romanofski) [reply src/UI/Actions.hs]: Yes. I'm all in.

File: src/UI/Actions.hs (hunk 1 above)
Comment (romanofski) [reply src/UI/Actions.hs]: Many thanks! Done

**Pull request merged**